libs/types/src/locations.rs

```diff
@@ -14,26 +14,17 @@ use cfg_primitives::AccountId;
 use frame_support::RuntimeDebugNoBound;
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
-use sp_core::crypto::AccountId32;
-// Please note that if this version change,
-// a migration could be required in those places where
-// RestrictedTransferLocation is stored
-use staging_xcm::v4::Location;
+use staging_xcm::VersionedLocation;
 
 use crate::domain_address::DomainAddress;
+
 /// Location types for destinations that can receive restricted transfers
 #[derive(Clone, RuntimeDebugNoBound, Encode, Decode, Eq, PartialEq, MaxEncodedLen, TypeInfo)]
 pub enum RestrictedTransferLocation {
 	/// Local chain account sending destination.
 	Local(AccountId),
 	/// XCM Location sending destinations.
-	Xcm(Location),
+	Xcm(Box<VersionedLocation>),
 	/// DomainAddress sending location from a liquidity pools' instance
 	Address(DomainAddress),
 }
-
-impl From<AccountId32> for RestrictedTransferLocation {
-	fn from(value: AccountId32) -> Self {
-		Self::Local(value)
-	}
-}
```

Comment:
Did you have an error with this? I remember runtime benchmarks not working without this.

Reply:
Let me check again with this enabled. I don't know why but recently I am often fooled by my cargo cache which I can only fix by cleaning completely. 

This change broke transfer allowlist benchmarks which required `T::AccountId <-> T::Location` conversion. Unfortunately, it was not possible to implement `From/TryFrom` for `AccountId` and `RestrictedTransferLocation` because of the the blanket Rust `From<T> for T` implementation which conflicts with any custom implementation involving the same type for both `From` and `TryFrom`. 

AFAICT, I could have either added a custom conversion trait or gone with my current proposal. Unfortunately, my proposal required refactoring unit tests which used `u64` as `AccountId` and a custom enum for `T::Location` which conflicted with my new benchmark restriction of `Config<AccountId = AccountId, Location = RestrictedTransferLocation`.

Reply:
> Unfortunately, it was not possible to implement From/TryFrom for AccountId and RestrictedTransferLocation

The above compiled in `main` IIRC 🤔

Reply:
Apparently, it works for `AccountId32` as part of our current codebase. However, it does not work for `AccountId` even though that's the same type under the hood 🤷

Reply:
Reverted in [a7cf00f](https://github.com/centrifuge/centrifuge-chain/pull/1861/commits/a7cf00f269ca79c562a91b7d36ebc0f625716fa5)

Reply:
Yeah, that is something I previously battled with. I do not understand why TBH. Even I was not able to find the `impl T for T` implementation running `cargo doc`. There is something weird in the implementation.

Reply:
I am happy to hear even you don't understand it. It really frustrated me yesterday..